provider/datagen/src/source.rs

```diff
@@ -177,6 +181,17 @@ impl SourceData {
         Self { collations, ..self }
     }
 
+    /// A path to a list_to_ucptrie binary (See the `icu_codepointtrie_builder` crate)
+    /// for building segmentation data. By default an internal version will be used, as long as
+    /// datagen is built with the `"wasm_cptbuilder"` feature.
+    pub fn with_list_to_ucptrie_binary(self, list_to_ucptrie_binary: PathBuf) -> Self {
```

Comment:
This has nothing to do with 'source data' and should be somewhere else (or just implicitly use the `PATH`).